### PR TITLE
Plugin setup

### PR DESCRIPTION
**Pull request by crusher083**
Plugin setup description for calculate GMHI function

### CODE REVIEW

File: q2_health_index/plugin_setup.py [hunk 1]
```diff
@@ -18,12 +19,18 @@
...
-    function=print_hello,
-    inputs={},
-    parameters={},
-    input_descriptions={},
-    parameter_descriptions={},
-    name='Print hello.',
-    description='Print hello on the screen.'
+plugin.methods.register_function(
+    function=calculate_gmhi,
+    inputs={'table': FeatureTable[RelativeFrequency],
```
Comment (tkosciol) [inline q2_health_index/plugin_setup.py:24]: `inputs={'table': FeatureTable[Frequency]` or am I missing something and it should be `RelativeFrequency`?

File: q2_health_index/plugin_setup.py [hunk 2]
```diff
@@ -18,12 +19,18 @@
...
+            'list': '.txt'},
```
Comment (tkosciol) [inline q2_health_index/plugin_setup.py:25]: I think this should be `TSVTaxonomyFormat`

File: q2_health_index/plugin_setup.py [hunk 3]
```diff
@@ -18,12 +19,18 @@
...
+    parameters={'metadata': Metadata},
+    outputs=[('gmhi_plot', Visualization),
+             ('gmhi_results', 'table of gmhi values for each sample')],
```
Comment (tkosciol) [inline q2_health_index/plugin_setup.py:28]: I would save ghmi_results as a `Metadata` for the ease of processing

File: q2_health_index/plugin_setup.py (hunk 1 above)
Comment (crusher083) [reply q2_health_index/plugin_setup.py:24]: In the paper they've used relative abundances, therefore we implemented data import through `FeatureTable[RelativeFrequency]`. This is already tested and working and will be added in the next commit.

File: q2_health_index/plugin_setup.py (hunk 2 above)
Comment (crusher083) [reply q2_health_index/plugin_setup.py:25]: Okay, I'll implement it and adapt plugin code in the next commit :)

Comment (crusher083) [reply q2_health_index/plugin_setup.py:25]: @tkosciol it worked as 

'species = Artifact.import_data('FeatureData[Taxonomy]', q2_h_species, view_type='HeaderlessTSVTaxonomyFormat')

def species_list_from_q2_artifact(feature_data_taxonomy: FeatureData[Taxonomy]) -> List[str]:
    df = feature_data_taxonomy.view(pd.DataFrame)
    species_list = [x for x in df['Taxon']]
    
    return species_list

mh_list = species_list_from_q2_artifact(species)'

File: q2_health_index/plugin_setup.py (hunk 1 above)
Comment (tkosciol) [reply q2_health_index/plugin_setup.py:25]: so just to be clear, `FeatureTable[RelativeFrequency]` has relative frequencies, ie. not counts. If we want to use raw Feature Tables at some point, they come as `FeatureTable[Frequency]` as they would need to be transformed before using.

Comment (crusher083) [reply q2_health_index/plugin_setup.py]: This makes sense, as they discard unclassified species, and then normalize the table. Good point, already corrected!

Comment (tkosciol) [reply q2_health_index/plugin_setup.py]: so just to be clear - that implies that the plugin would have another method that takes raw feature table and transforms it to relative frequency table. One way to go about it then, would be to have 2 methods and 1 pipeline:
methods:
* transform `FeatureTable[Frequency]` to `FeatureTable[RelativeFrequency]`
* calculate the GMHI on `FeatureTable[RelativeFrequency]`
plpeline:
* do both from the above

**Comment by tkosciol** [comment]
I can merge if you're ok with my comments

**Comment by crusher083** [comment]
Sure, I corrected everything :)

**Pull request merged**